modules/database/src/ioc/dbStatic/dbLexRoutines.c

```diff
@@ -159,9 +159,10 @@ static void *getLastTemp(void)
     return(ptempListNode->item);
 }
 
-const char *dbOpenFile(DBBASE *pdbbase,const char *filename,FILE **fp)
+const char *dbOpenFile(DBBASE **ppdbbase,const char *filename,FILE **fp)
 {
-    ELLLIST     *ppathList = (ELLLIST *)pdbbase->pathPvt;
+    if (*ppdbbase == 0) *ppdbbase = dbAllocBase();
+    ELLLIST     *ppathList = (ELLLIST *)(*ppdbbase)->pathPvt;
```

Comment:
In plain C code, some older compilers still that variable definitions be grouped at the top of each block.  So in this case:

```c
    ELLLIST     *ppathList;
    if (*ppdbbase == 0) *ppdbbase = dbAllocBase();
    ppathList = (ELLLIST *)(*ppdbbase)->pathPvt;
```